core/src/main/scala/tofu/syntax/feither.scala

```diff
@@ -0,0 +1,210 @@
+
+  implicit final class EitherFOps[F[_], L, R](private val e: F[Either[L, R]]) extends AnyVal {
+
+    def orElseF[L1 >: L, R1 >: R](f: => F[Either[L1, R1]])(implicit F: Monad[F]): F[Either[L1, R1]] = {
+      F.flatMap(e) {
+        case r @ Right(_) => F.pure(r.leftCast)
+        case _            => f
+      }
+    }
+
+    def getOrElseF[R1 >: R](f: => F[R1])(implicit F: Monad[F]): F[R1] = {
+      F.flatMap(e) {
+        case Right(value) => F.pure(value)
+        case _            => f
+      }
+    }
+
+    def catchAll[R1 >: R](f: L => F[R1])(implicit F: Monad[F]): F[R1] = {
+      F.flatMap(e) {
+        case Left(err)    => f(err)
+        case Right(right) => F.pure(right)
+      }
+    }
+
+    def absolve[R1 >: R](implicit R: Raise[F, L], F: Monad[F]): F[R1] = {
+      F.flatMap(e) {
+        case Right(right) => F.pure(right)
+        case Left(left)   => R.raise(left)
+      }
+    }
+
+    def assocR[A, B](implicit F: Functor[F], ev: R <:< Either[A, B]): F[Either[Either[L, A], B]] = {
+      F.map(e) {
+        case Right(r) =>
+          ev(r) match {
+            case Left(a)      => Left(Right(a))
+            case b @ Right(_) => b.leftCast
+          }
+        case left @ Left(_) => left.rightCast.asLeft
+      }
+    }
+
+    def assocL[A, B](implicit F: Functor[F], ev: L <:< Either[A, B]): F[Either[A, Either[B, R]]] = {
+      F.map(e) {
+        case r @ Right(_) => r.leftCast.asRight
+        case Left(l) =>
+          ev(l) match {
+            case left @ Left(_) => left.rightCast
+            case Right(b)       => b.asLeft.asRight
+          }
+      }
+    }
+
+    def mapF[B](f: R => F[B])(implicit F: Monad[F]): F[Either[L, B]] = {
+      F.flatMap(e)(_.traverse(f))
+    }
+
+    def leftMapF[L1](f: L => F[L1])(implicit F: Monad[F]): F[Either[L1, R]] = {
+      F.flatMap(e) {
+        case Left(left)       => F.map(f(left))(_.asLeft)
+        case right @ Right(_) => F.pure(right.leftCast)
+      }
+    }
+
+    def flatMapIn[L1 >: L, B](f: R => Either[L1, B])(implicit F: Functor[F]): F[Either[L1, B]] = {
+      F.map(e)(_.flatMap(f))
+    }
+
+    def leftFlatMapIn[L1, R1 >: R](f: L => Either[L1, R1])(implicit F: Functor[F]): F[Either[L1, R1]] = {
+      F.map(e)(_.leftFlatMap(f))
+    }
+
+    def flatMapF[L1 >: L, B](f: R => F[Either[L1, B]])(implicit F: Monad[F]): F[Either[L1, B]] = {
+      F.flatMap(e) {
+        case Right(right)   => f(right)
+        case left @ Left(_) => F.pure(left.rightCast)
+      }
+    }
+
+    def leftFlatMapF[R1 >: R, L1](f: L => F[Either[L1, R1]])(implicit F: Monad[F]): F[Either[L1, R1]] = {
+      F.flatMap(e) {
+        case Left(left)       => f(left)
+        case right @ Right(_) => F.pure(right.leftCast)
+      }
+    }
+
+    def doubleFlatMap[L1 >: L, R1 >: R](f: R => F[Either[L1, R1]])(implicit F: Monad[F]): F[Either[L1, R1]] = {
+      F.flatMap(e) {
+        case Right(right)   => f(right)
+        case left @ Left(_) => F.pure(left.rightCast)
+      }
```

Comment:
I think we can keep just one method from doubleFlatMap, flatMapF

Reply:
Ok, seems good